recipes/opencv/4.x/conanfile.py

```diff
@@ -263,6 +267,11 @@ def _configure_cmake(self):
 
     def build(self):
         self._patch_opencv()
+
+        if self.options.with_jpeg2000 == "jasper":
+            if tools.Version(self.version) < "4.5.0" and self.deps_cpp_info["jasper"].version > "2.0.16":
+                raise ConanInvalidConfiguration("OpenCV < 4.5.0 needs an older jasper version, see https://github.com/opencv/opencv/issues/17984")
```

Comment:
Wouldn't it be better to patch opencv versions < 4.5.0 in this recipe? Seems to be a tiny patch: https://github.com/opencv/opencv/pull/17983, and it will avoid dependency graph conflicts.

Reply:
only > 2.0.16

Reply:
good idea, using a patch now

Reply:
Yes, I feel that all jasper versions should work with this patch because now they call `jas_matrix_getref` instead of `jas_matrix_get`